### PR TITLE
add num_inference_steps arg to DDPM

### PR DESCRIPTION
**Pull request by tmabraham**
Before it was set to a default of 1000. I think it's fair to give users the option to choose their own number of inference steps like any of the other pipelines but if there is a good reason not to, feel free to close.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@anton-l @patil-suraj what do you think - I don't have strong feeling here tbh

File: src/diffusers/pipelines/ddpm/pipeline_ddpm.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ def __call__(
         self,
         batch_size: int = 1,
         generator: Optional[torch.Generator] = None,
+        num_inference_steps: int = 1000,
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/ddpm/pipeline_ddpm.py:45]:
```suggestion
        num_inference_steps: Optional[int] = None,
```

**Review by patrickvonplaten — APPROVED** [review]
Ok for me! Thanks for the PR

**Review by patil-suraj — APPROVED** [review]
Good for me, thanks for the PR @tmabraham

**Pull request merged**